File: backend/src/main/java/ai/applysmart/config/FeatureFlags.java
package ai.applysmart.config;

/**
 * Feature Flags Configuration
 *
 * Centralized feature flag management for toggling features on/off
 * without code changes in production.
 */
public final class FeatureFlags {

    /**
     * Subscription & Billing Features
     * Controls all subscription-related functionality including:
     * - Subscription plans and restrictions
     * - Billing endpoints
     * - Plan-based access control
     */
    public static final boolean SUBSCRIPTIONS_ENABLED = false;

    /**
     * Admin Panel Subscription Analytics
     * Controls subscription-related data in admin dashboard
     */
    public static final boolean ADMIN_SUBSCRIPTION_ANALYTICS = false;

    // Private constructor to prevent instantiation
    private FeatureFlags() {
        throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
    }
}

File: frontend/src/shared/config/feature-flags.ts
export const FEATURE_FLAGS = {
  SUBSCRIPTIONS_ENABLED: false,
  ADMIN_SUBSCRIPTION_ANALYTICS: false,
} as const;

export const isFeatureEnabled = (feature: keyof typeof FEATURE_FLAGS): boolean => {
  return FEATURE_FLAGS[feature];
};

export type FeatureFlag = keyof typeof FEATURE_FLAGS;
